wp_api_integration_tests/tests/test_cookie_authentication_immut.rs

```diff
@@ -0,0 +1,70 @@
+use wp_api::{WpApiClient, WpAuthentication, WpLoginCredentials};
+
+use wp_api_integration_tests::*;
+
+use serial_test::serial;
+
+#[tokio::test]
+#[serial]
+async fn test_cookie_authentication() {
+    let login = WpLoginCredentials {
+        username: TestCredentials::instance().admin_username.to_string(),
+        password: TestCredentials::instance()
+            .admin_account_password
+            .to_string(),
+    };
+    let request_executor = std::sync::Arc::new(AsyncWpNetworking::with_cookie_store());
+    let client = WpApiClient::new(
+        test_site_url(),
+        WpAuthentication::UserAccount { login },
+        request_executor,
+    );
+    let user = client
+        .users()
+        .retrieve_me_with_edit_context()
+        .await
+        .assert_response();
+    assert_eq!(user.id, FIRST_USER_ID);
+}
+
+#[tokio::test]
+#[serial]
```

Comment:
```suggestion
#[parallel]
```

The reason we use `#[serial]` in mutable tests is because the execution of a mutable test can interfere with other tests - both mutable and immutable ones. However, the same doesn't apply for immutable tests, and it's safe to run them in parallel.